cli/tests/program.rs

```diff
@@ -1614,3 +1825,176 @@ fn test_cli_program_dump() {
     }
 }
+
+fn fetch_pre_signer(cmd_output_json_str: &str, target_pub_key: &str) -> Presigner {
+    let json: Value = serde_json::from_str(cmd_output_json_str).unwrap();
+    let target_pre_signer_str = json
+        .as_object()
+        .unwrap()
+        .get("signers")
+        .unwrap()
+        .as_array()
+        .unwrap()
+        .iter()
+        .find(|pre_signer| pre_signer.as_str().unwrap().starts_with(target_pub_key))
+        .unwrap()
+        .as_str()
+        .unwrap();
+    let mut parts = target_pre_signer_str.split('=');
+    assert_eq!(2, parts.clone().count());
+    Presigner::new(
+        &Pubkey::from_str(parts.next().unwrap()).unwrap(),
+        &Signature::from_str(parts.next().unwrap()).unwrap(),
+    )
+}
+
+fn create_buffer_with_offline_authority<'a>(
+    rpc_client: &RpcClient,
+    program_path: &Path,
+    config: &mut CliConfig<'a>,
+    online_signer: &'a Keypair,
+    offline_signer: &'a Keypair,
+    buffer_signer: &'a Keypair,
+    buffer_signer_identity: &'a NullSigner,
+) {
+    // Write a buffer
+    config.signers = vec![online_signer, buffer_signer];
+    config.command = CliCommand::Program(ProgramCliCommand::WriteBuffer {
+        program_location: program_path.to_str().unwrap().to_string(),
+        fee_payer_signer_index: 0,
+        buffer_signer_index: Some(1),
+        buffer_pubkey: Some(buffer_signer.pubkey()),
+        buffer_authority_signer_index: 0,
+        max_len: None,
+        skip_fee_check: false,
+    });
+    process_command(config).unwrap();
+    let buffer_account = rpc_client.get_account(&buffer_signer.pubkey()).unwrap();
+    if let UpgradeableLoaderState::Buffer { authority_address } = buffer_account.state().unwrap() {
+        assert_eq!(authority_address, Some(online_signer.pubkey()));
+    } else {
+        panic!("not a buffer account");
+    }
+
+    // Set buffer authority to offline signer
+    config.signers = vec![online_signer, buffer_signer];
+    config.command = CliCommand::Program(ProgramCliCommand::SetBufferAuthority {
+        buffer_pubkey: buffer_signer.pubkey(),
+        buffer_authority_index: Some(0),
+        new_buffer_authority: offline_signer.pubkey(),
+    });
+    config.output_format = OutputFormat::JsonCompact;
+    let response = process_command(config);
+    let json: Value = serde_json::from_str(&response.unwrap()).unwrap();
+    let offline_signer_authority_str = json
+        .as_object()
+        .unwrap()
+        .get("authority")
+        .unwrap()
+        .as_str()
+        .unwrap();
+    assert_eq!(
+        Pubkey::from_str(offline_signer_authority_str).unwrap(),
+        offline_signer.pubkey()
+    );
+    let buffer_account = rpc_client.get_account(&buffer_signer.pubkey()).unwrap();
+    if let UpgradeableLoaderState::Buffer { authority_address } = buffer_account.state().unwrap() {
+        assert_eq!(authority_address, Some(offline_signer.pubkey()));
+    } else {
+        panic!("not a buffer account");
+    }
```

Comment:
nit-up-to-you: this is already being checked in other tests, so it's not necessary to do the checks here

Reply:
Mostly doing it "to fail fast" (in case I'm running this test in isolation I'll likely have hard time debugging the failure reason), so I'd rather have extra checks in palace than spend more time debugging and such :)